src/hotspot/share/utilities/debug.cpp

```diff
@@ -636,8 +637,41 @@ extern "C" JNIEXPORT void findpc(intptr_t x) {
   os::print_location(tty, x, true);
 }
 
+// flags must be OR'ed from ClassPrinter::Mode for the next 5 functions.
+// Examples (in gdb):
+//   call findclass("java/lang/Object", 0x3)
+//   call findmethod("*ang/Object*", "wait", 0x7)
+//   call findmethod2("*ang/Object*", "wait", "(J*", 0x1)
+extern "C" JNIEXPORT void findclass(const char* class_name_pattern, int flags) {
+  Command c("findclass");
+  ClassPrinter::print_classes(class_name_pattern, flags, tty);
+}
+
+extern "C" JNIEXPORT void findmethod(const char* class_name_pattern,
+                                     const char* method_name_pattern, int flags) {
+  Command c("findmethod");
+  ClassPrinter::print_methods(class_name_pattern, method_name_pattern, flags, tty);
+}
+
+extern "C" JNIEXPORT void findmethod2(const char* class_name_pattern,
+                                      const char* method_name_pattern,
+                                      const char* method_signature_pattern, int flags) {
+  Command c("findmethod2");
+  ClassPrinter::print_methods(class_name_pattern, method_name_pattern,
+                              method_signature_pattern, flags, tty);
+}
+
+extern "C" JNIEXPORT void printclass(intptr_t k, int flags) {
```

Comment:
It's weird that the function to print the method is findmethod, and the one to print the class is printclass. I see that findmethod is consistent with the 'find' functions above.  Maybe this should be findclass() too.

Reply:
I had two groups of printing functions with the following intention:

- findclass/findmethod will search for class/methods by their names. Call these when you don't have an InstanceKlass* or Method*
- printclass/printmethod should be used when you already have an InstanceKlass* or Method*

However, I realized that the second group of methods should really be something like InstanceKlass::print_xxx() and Method::print_xxx(), so you can do something like this in gdb 

```
call ik->print_all_methods_with_bytecodes();
```

I have removed the second group of functions from this PR (commit [6675901](https://github.com/openjdk/jdk/pull/9957/commits/6675901125018853bad316a62702a8f7c62f331b)). I may add them in a follow-up PR if necessary.

Reply:
Yes, I agree.  If you already have the InstanceKlass, it should be a function using that as a 'this' pointer.